src/persist-client/src/internal/trace.rs

```diff
@@ -1289,45 +1256,32 @@ impl<T: Timestamp + Lattice> Spine<T> {
 
                 // Continue only as far as is appropriate
                 while appropriate_level < length - 1 {
-                    match self.merging[length - 2].take() {
-                        // Vacant batches can be absorbed.
-                        MergeState::Vacant => {
-                            self.merging.remove(length - 2);
-                            length = self.merging.len();
-                        }
-                        // Single batches may initiate a merge, if sizes are
-                        // within bounds, but terminate the loop either way.
-                        MergeState::Single(batch) => {
+                    let current = &mut self.merging[length - 2];
+                    if current.is_vacant() {
+                        self.merging.remove(length - 2);
+                        length = self.merging.len();
+                    } else {
+                        if !current.is_full() {
```

Comment:
I experimented with this, but since this is the only place we could use the enum, it it ended up making the diff less clear overall. Will definitely keep this in mind as we make more changes though!